src/app/searchBar.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from "react";
+import { AutoComplete } from "antd";
+import { SearchOutlined } from "@ant-design/icons";
+import { Button } from "antd";
+
+const SearchBar: React.FC<{
+  search: string;
+  setSearch: any;
+  onClick: () => void;
+  onKeyDown: any;
+}> = ({ search, setSearch, onClick, onKeyDown }) => {
+  const [searchData, setSearchData] = useState<{ value: string }[]>([]);
+
+  const handleChange = (e: string) => {
+    setSearch(e);
+  };
+
+  useEffect(() => {
+    fetch("/world_cities.json")
+      .then((res) => res.json())
+      .then((data) => {
+        setSearchData(data);
+      });
+  }, []);
+
+  const getPanelValue = () => {
+    return searchData
+      .filter((item: any, index: any) => {
+        return item.name.toLowerCase().startsWith(search.toLowerCase());
+      })
+      .map((item: any, index: any) => {
+        return { value: item.name };
+      });
+  };
+
+  const onSelect = (data: string) => {
+    console.log("onSelect", data);
+  };
+
+  return (
+    <>
+      <AutoComplete
+        className="search_bar"
+        id="search_bar"
+        options={getPanelValue()}
```

Comment:
Generating options with every search change might not be intended use and performance killer.
Probably a better approach can be achieved with filterOption and other AutoComplete property.